lib/strutils.c

```diff
@@ -1217,6 +1217,27 @@ int ul_optstr_next(char **optstr, char **name, size_t *namesz,
 	return 1;				/* end of optstr */
 }
 
+int errsnprint(char *buf, size_t bufsz, int errnum, const char *fmt, ...)
```

Comment:
This seems weird.

The advantage of `"%m"` is that it saves a `"%s", strerror(errno)`.
It is non-standard and of limited advantage.

Instead of introducing a custom helper to do the same with arbitrary `errnum` the user could just specify `"%s", strerror(errnum)`.

(Also it should probably be called `errsnprintf`.